src/Tmds.Ssh/ChannelContextReceiveMessageExtensions.cs

```diff
@@ -67,5 +68,40 @@ static void ParseChannelOpenConfirmation(ReadOnlyPacket packet, string failureMe
                 }
             }
         }
+
+        public static async ValueTask<int> ReceiveServerVersionAsync(this ChannelContext context, string failureMessage, CancellationToken ct)
+        {
+            using var packet = await context.ReceivePacketAsync(ct).ConfigureAwait(false);
+
+            return ParseSftpVersion(packet, failureMessage);
+            /*
+                        byte            SSH_MSG_CHANNEL_DATA
+                        uint32          recipient channel
+                        string          data
+
+                        uint32          SftpLength
+                        byte            SftpType
+                        uint32          SftpVersion
+                        string          extension-name
+                        string          extension-data
+            */
+
+            static int ParseSftpVersion(ReadOnlyPacket packet, string failureMessage)
+            {
+                var reader = packet.GetReader();
+                reader.ReadMessageId(MessageId.SSH_MSG_CHANNEL_DATA);
+                // Also maybe deal with multiple packets sent with the VERSION packet?
+                // var channelId = reader.ReadUInt32();
+                // var dataLength = reader.ReadUInt32();
+                // var sftpPacketLength = reader.ReadUInt32();
+                reader.Skip(12);
+                var type = (SftpPacketType)reader.ReadByte();
+                if (type != SftpPacketType.SSH_FXP_VERSION)
+                    ThrowHelper.ThrowProtocolUnexpectedSftpPacketType(SftpPacketType.SSH_FXP_VERSION);
```

Comment:
Let's add a method to the reader, similar to the implementation of `ReadMessageId`.